EIPS/eip-4742.md

```diff
@@ -0,0 +1,56 @@
+---
+eip: 4742
+title: NONCE opcode
+description: A new EVM opcode to get the current nonce of an address.
+author: 0xfoobar (@0xfoobar)
+discussions-to: https://ethereum-magicians.org/t/eip-4672-account-nonce-opcode/8171
+status: Draft
+type: Standards Track
+category: Core
+created: 2022-02-01
+---
+
+## Abstract
+
+We propose an additional EVM opcode, `NONCE` (0x47), which could provide novel protection against sybil attacks and expand the smart contract design space.
+
+## Motivation
+
+NFT contracts commonly use checks on `msg.sender` to limit sybil attacks. However, this is easily circumvented by spinning up new addresses, or even nesting address creation within a smart contract. Letting smart contracts check `NONCE` would add additional costs to this, serving as a “proof of history”. While not bulletproof (as msg.sender is not), it opens up the design space to creative uses.
+
+## Specification
+
+A new opcode, `NONCE`, is introduced, with number `0x47`. The `NONCE` opcode takes one argument from the stack, and pushes to the stack the nonce corresponding to that address.
+
+The gas cost of the `NONCE` is 200.
+
+
+## Rationale
+
+The EVM makes several global variables available, examples include `block.timestamp`, `block.number`, `block.difficulty`, `msg.sender`, and `gasleft()`. It also provides several context-dependent opcodes, such as `EXTCODESIZE`, `BASEFEE`, and `BALANCE` Smart contracts can leverage these info for novel techniques, such as staking pool emissions, limiting flashloan attacks, or NFTs that embed blockchain state into their mint result.
+
+It is useful to gather information about interacting addresses. NFT minting is an immediate usecase, and it follows that as much information about the underlying blockchain state should be made available to smart contracts.
+
+## Backwards Compatibility
+
+There are no backwards compatibility concerns.
+
+## Test Cases
+
+1. The `NONCE` of an address with no transactions is `0`.
+2. The `NONCE` of an externally owned address (EOA) with one originating transaction is `1`.
+3. The `NONCE` of a smart contract, deployed after EIP 161, that has not deployed any other smart contracts is `1`.
+4. The `NONCE` of a smart contract, deployed before EIP 161, that has not deployed any other smart contracts is `0`.
+5. The `NONCE` of a smart contract, deployed after EIP 161, that has deployed one other smart contract is `2`.
+6. The `NONCE` of a smart contract, deployed before EIP 161, that has deployed one other smart contract is `1`.
+
+
+## Implementation
+
+Not provided yet, would need to be integrated into Ethereum clients as this is an EVM modification. However, clients already check the account nonce on each transaction to ensure validity, so exposing this data as an opcode should not add significant processing or development load.
+
+
```

Comment:
```suggestion
```
There is no Implementation section.  There is a Reference Implementation section but it is optional so if you don't have one it is best to just leave it out.